kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/extension/ClientAdapter.java

```diff
@@ -36,19 +35,15 @@
 /**
  * To be used as the base class for creating extension clients
  */
-public abstract class ClientAdapter<C extends Client> implements Client {
+public abstract class ClientAdapter<C extends ClientAdapter<C>> implements Client {
```

Comment:
In OperationContext a new method
```java
  <T extends HasMetadata, V extends VisitableBuilder<T, V>> ResourceHandler<T, V> getHandler(T item) {
    return getClient().adapt(BaseClient.class).getHandlers().get(item, getClient());
  }
```

Is this something we might want to replicate here so that we can reuse in the DSL clients?

```java
  Handlers getHandlers() {
    return getClient().adapt(BaseClient.class).getHandlers();
  }
```

or even:

```java
... getNamespacedHasMetadataCreateOnlyOperation(...) {}
... getNonListingOperation(...) {}
```

Or maybe it's just not worth it?

Reply:
> Is this something we might want to replicate here so that we can reuse in the DSL clients?

I'm trying to not expose Handlers to the api.  There are only 2 usages of getHandler - to get a Builder, and to create an operation in the same context as the current.

For the Builder I'd much prefer updating the model generation logic to do something similar to lombok - add a asBuilder() method / interface to each class.  Only if the resource does not implement Buildable, would you try to lookup the builder via the classpath.

Creating an operation in the same context as the current - that is only currently needed by the resourceList operations as you can do resourceList(...).dryRun().resources().  With the ExtensibleResourceAdapter that would only be needed if you override the behavior of a mutating method (create, edit, etc.) in a way that wanted to do a modification on an outside resource (none of the logic does this yet).  If you just add a new method they are not available to be called after a dsl method like dryRun.